packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -69,28 +76,62 @@ android {
 <% if (project.arch === "new") { -%>
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
 <% } -%>
+<% if (project.cpp || project.nitro) { -%>
 
-<% if (project.cpp) { -%>
     externalNativeBuild {
       cmake {
-        cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
+        cppFlags "-frtti -fexceptions -Wall -fstack-protector-all"
+        arguments "-DANDROID_STL=c++_shared"
         abiFilters (*reactNativeArchitectures())
+
+        buildTypes {
+          debug {
+            cppFlags "-O1 -g"
+          }
+          release {
+            cppFlags "-O2"
+          }
+        }
```

Comment:
If these are only needed for nitro make sure to wrap it a condition

Reply:
These are nice to have for every type of C++ library so let's definitely keep them. An overall improvement.